policy/modules/system/systemd.if

```diff
@@ -58,28 +60,612 @@ template(`systemd_role_template',`
 	# This domain is per-role because of the below transitions.
 	# See the systemd --user section of systemd.te for the
 	# remainder of the rules.
-	allow $1_systemd_t $3:process { setsched rlimitinh };
+	allow $1_systemd_t $3:process { setsched rlimitinh signal_perms };
+	allow $1_systemd_t self:netlink_kobject_uevent_socket create_socket_perms;
 	corecmd_shell_domtrans($1_systemd_t, $3)
 	corecmd_bin_domtrans($1_systemd_t, $3)
-	allow $1_systemd_t self:process signal;
+	allow $1_systemd_t self:process { getsched signal };
+
+	dontaudit $3 $1_systemd_t:unix_stream_socket rw_stream_socket_perms;
 
 	files_search_home($1_systemd_t)
 
+	init_link_keyring($1_systemd_t)
+
+	miscfiles_watch_localization($1_systemd_t)
+
+	mount_read_runtime_files($1_systemd_t)
+	mount_watch_runtime_files($1_systemd_t)
+	mount_watch_reads_runtime_files($1_systemd_t)
+
+	allow $3 $1_systemd_t:process signal;
 	# Allow using file descriptors for user environment generators
 	allow $3 $1_systemd_t:fd use;
 	allow $3 $1_systemd_t:fifo_file rw_inherited_fifo_file_perms;
 
+	allow $1_systemd_t self:unix_stream_socket rw_socket_perms;
```

Comment:
This should probably be just `create_stream_socket_perms` because systemd --user accepts unix stream connections on its /run/user/*/systemd/private and notify sockets